### DISCUSSION TITLE
Open Interest write to Postgres

### DISCUSSION THREAD

**Discussion by joshuaspear** [bmoscon/cryptofeed#691] [Q&A] [answered]
Hi, 

I'm having difficulty using the Postgres open interest backend. The class OpenInterestPostgres defines a write method which subsequently calls the write method defined in the parent class PostgresCallback however, I am not sure if it has been implemented correctly. The PostgresCallback write method excepts only parameter "data" however, the OpenInterestPostgres write method supplied several parameters to it which I believe is causing issues in the program I am writing. 

Please can I confirm whether this is a bug or whether I have miss understood the use of the class? If it is the latter, please could I have some guidance on how to use it. 

Thanks

**Comment by bmoscon** [comment, marked as answer]
@joshuaspear - seems like a bug, can you open a ticket? All the backend code changed recently, and I'm only able to manually retest so much every time. At some point I want to spend the time generating integration tests for all the backends, but thats a massive undertaking as well.

**Reply by joshuaspear** [reply]
@bmoscon many thanks for the reply and appreciate how bigger project this is. Thank you for putting it together
